crypto/pem/pem_lib.c

```diff
@@ -785,13 +784,37 @@ int PEM_read_bio(BIO *bp, char **name, char **header, unsigned char **data,
 }
 
 int PEM_def_callback(char *buf, int size, int rwflag, void *userdata) {
-  if (!buf || !userdata || size < 0) {
+  if (!buf || size < 0) {
```

Comment:
* Fail on `size <= 0`?
* Since this is a public function, should we proactively zero the `buf` to help prevent its current contents from being considered password were a failure to go unchecked by the caller?